File: old/tags/2.2/cpp/DIS/SystemID.cpp
#include <DIS/SystemID.h> 

using namespace DIS;


SystemID::SystemID():
   _systemType(0), 
   _systemName(0), 
   _systemMode(0), 
   _changeOptions(0)
{
}

SystemID::~SystemID()
{
}

unsigned short SystemID::getSystemType() const
{
    return _systemType;
}

void SystemID::setSystemType(unsigned short pX)
{
    _systemType = pX;
}

unsigned short SystemID::getSystemName() const
{
    return _systemName;
}

void SystemID::setSystemName(unsigned short pX)
{
    _systemName = pX;
}

unsigned char SystemID::getSystemMode() const
{
    return _systemMode;
}

void SystemID::setSystemMode(unsigned char pX)
{
    _systemMode = pX;
}

unsigned char SystemID::getChangeOptions() const
{
    return _changeOptions;
}

void SystemID::setChangeOptions(unsigned char pX)
{
    _changeOptions = pX;
}

void SystemID::marshal(DataStream& dataStream) const
{
    dataStream << _systemType;
    dataStream << _systemName;
    dataStream << _systemMode;
    dataStream << _changeOptions;
}

void SystemID::unmarshal(DataStream& dataStream)
{
    dataStream >> _systemType;
    dataStream >> _systemName;
    dataStream >> _systemMode;
    dataStream >> _changeOptions;
}


bool SystemID::operator ==(const SystemID& rhs) const
 {
     bool ivarsEqual = true;

     if( ! (_systemType == rhs._systemType) ) ivarsEqual = false;
     if( ! (_systemName == rhs._systemName) ) ivarsEqual = false;
     if( ! (_systemMode == rhs._systemMode) ) ivarsEqual = false;
     if( ! (_changeOptions == rhs._changeOptions) ) ivarsEqual = false;

    return ivarsEqual;
 }

int SystemID::getMarshalledSize() const
{
   int marshalSize = 0;

   marshalSize = marshalSize + 2;  // _systemType
   marshalSize = marshalSize + 2;  // _systemName
   marshalSize = marshalSize + 1;  // _systemMode
   marshalSize = marshalSize + 1;  // _changeOptions
    return marshalSize;
}

File: old/tags/2.2/java/edu/nps/moves/dis/SystemID.java
package edu.nps.moves.dis;

import java.util.*;
import java.io.*;
import edu.nps.moves.jaxb.dis.*;

/**
 * 5.2.58. Used in IFF ATC PDU
 *
 * Copyright (c) 2007, MOVES Institute, Naval Postgraduate School. All rights reserved.
 * This work is licensed under the BSD open source license, available at https://www.movesinstitute.org/licenses/bsd.html
 *
 * @author DMcG
 */
public class SystemID extends Object implements Serializable
{
   /** System Type */
   protected int  systemType;

   /** System name, an enumeration */
   protected int  systemName;

   /** System mode */
   protected short  systemMode;

   /** Change Options */
   protected short  changeOptions;


/** Constructor */
 public SystemID()
 {
 }

/** 
 * Constructor--takes a parallel jaxb object and returns an open-dis object 
 * 1.4_sed_bait_start */
 public SystemID(edu.nps.moves.jaxb.dis.SystemID x)
 {
     this.systemType = x.getSystemType();
     this.systemName = x.getSystemName();
     this.systemMode = x.getSystemMode();
     this.changeOptions = x.getChangeOptions();
 }
/* 1.4_sed_bait_end */


/**
 * returns a jaxb object intialized from this object, given an empty jaxb object
 * 1.4_sed_bait_start **/
 public edu.nps.moves.jaxb.dis.SystemID initializeJaxbObject(edu.nps.moves.jaxb.dis.SystemID x)
 {
     ObjectFactory factory = new ObjectFactory();

     x.setSystemType( this.getSystemType() );
     x.setSystemName( this.getSystemName() );
     x.setSystemMode( this.getSystemMode() );
     x.setChangeOptions( this.getChangeOptions() );
   return x;
 }
/* 1.4_sed_bait_end */


public int getMarshalledSize()
{
   int marshalSize = 0; 

   marshalSize = marshalSize + 2;  // systemType
   marshalSize = marshalSize + 2;  // systemName
   marshalSize = marshalSize + 1;  // systemMode
   marshalSize = marshalSize + 1;  // changeOptions

   return marshalSize;
}


public void setSystemType(int pSystemType)
{ systemType = pSystemType;
}

public int getSystemType()
{ return systemType; 
}

public void setSystemName(int pSystemName)
{ systemName = pSystemName;
}

public int getSystemName()
{ return systemName; 
}

public void setSystemMode(short pSystemMode)
{ systemMode = pSystemMode;
}

public short getSystemMode()
{ return systemMode; 
}

public void setChangeOptions(short pChangeOptions)
{ changeOptions = pChangeOptions;
}

public short getChangeOptions()
{ return changeOptions; 
}


public void marshal(DataOutputStream dos)
{
    try 
    {
       dos.writeShort( (short)systemType);
       dos.writeShort( (short)systemName);
       dos.writeByte( (byte)systemMode);
       dos.writeByte( (byte)changeOptions);
    } // end try 
    catch(Exception e)
    { 
      System.out.println(e);}
    } // end of marshal method

public void unmarshal(DataInputStream dis)
{
    try 
    {
       systemType = dis.readShort();
       systemName = dis.readShort();
       systemMode = dis.readByte();
       changeOptions = dis.readByte();
    } // end try 
   catch(Exception e)
    { 
      System.out.println(e); 
    }
 } // end of unmarshal method 


 /**
  * The equals method doesn't always work--mostly on on classes that consist only of primitives. Be careful.
  */
 public boolean equals(SystemID rhs)
 {
     boolean ivarsEqual = true;

    if(rhs.getClass() != this.getClass())
        return false;

     if( ! (systemType == rhs.systemType)) ivarsEqual = false;
     if( ! (systemName == rhs.systemName)) ivarsEqual = false;
     if( ! (systemMode == rhs.systemMode)) ivarsEqual = false;
     if( ! (changeOptions == rhs.changeOptions)) ivarsEqual = false;

    return ivarsEqual;
 }
} // end of class
